File: MPChartExample/src/main/java/com/xxmassdeveloper/mpchartexample/notimportant/MainActivity.java

package com.xxmassdeveloper.mpchartexample.notimportant;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.view.WindowManager;
import android.widget.AdapterView;
import android.widget.AdapterView.OnItemClickListener;
import android.widget.ListView;

import com.github.mikephil.charting.utils.Utils;
import com.xxmassdeveloper.mpchartexample.AnotherBarActivity;
import com.xxmassdeveloper.mpchartexample.BarChartActivity;
import com.xxmassdeveloper.mpchartexample.BarChartActivityMultiDataset;
import com.xxmassdeveloper.mpchartexample.BarChartActivitySinus;
import com.xxmassdeveloper.mpchartexample.BarChartPositiveNegative;
import com.xxmassdeveloper.mpchartexample.BubbleChartActivity;
import com.xxmassdeveloper.mpchartexample.CandleStickChartActivity;
import com.xxmassdeveloper.mpchartexample.CombinedChartActivity;
import com.xxmassdeveloper.mpchartexample.CubicLineChartActivity;
import com.xxmassdeveloper.mpchartexample.DynamicalAddingActivity;
import com.xxmassdeveloper.mpchartexample.FilledLineActivity;
import com.xxmassdeveloper.mpchartexample.HalfPieChartActivity;
import com.xxmassdeveloper.mpchartexample.HighlightBubbleChart;
import com.xxmassdeveloper.mpchartexample.HorizontalBarChartActivity;
import com.xxmassdeveloper.mpchartexample.InvertedLineChartActivity;
import com.xxmassdeveloper.mpchartexample.LineChartActivity1;
import com.xxmassdeveloper.mpchartexample.LineChartActivityColored;
import com.xxmassdeveloper.mpchartexample.LineChartTime;
import com.xxmassdeveloper.mpchartexample.ListViewBarChartActivity;
import com.xxmassdeveloper.mpchartexample.ListViewMultiChartActivity;
import com.xxmassdeveloper.mpchartexample.MultiLineChartActivity;
import com.xxmassdeveloper.mpchartexample.PerformanceLineChart;
import com.xxmassdeveloper.mpchartexample.PieChartActivity;
import com.xxmassdeveloper.mpchartexample.PiePolylineChartActivity;
import com.xxmassdeveloper.mpchartexample.R;
import com.xxmassdeveloper.mpchartexample.RadarChartActivity;
import com.xxmassdeveloper.mpchartexample.RealtimeLineChartActivity;
import com.xxmassdeveloper.mpchartexample.ScatterChartActivity;
import com.xxmassdeveloper.mpchartexample.ScrollViewActivity;
import com.xxmassdeveloper.mpchartexample.StackedBarActivity;
import com.xxmassdeveloper.mpchartexample.StackedBarActivityNegative;
import com.xxmassdeveloper.mpchartexample.fragments.SimpleChartDemo;

import java.util.ArrayList;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity implements OnItemClickListener {
    ArrayList<ContentItem> mActivities = new ArrayList<>();
    int mPosition = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN,
                WindowManager.LayoutParams.FLAG_FULLSCREEN);
        setContentView(R.layout.activity_main);

        setTitle("MPAndroidChart Example");

        // initialize the utilities
        Utils.init(this);

        addHeader("Highlight Axes");
        addExample("Bubble Chart", "Highlight in two lines of code", HighlightBubbleChart.class);

        addHeader("Line Charts");

        addExample("Basic", "Simple line chart.", LineChartActivity1.class);
        addExample("Multiple", "Show multiple data sets.", MultiLineChartActivity.class);
        addExample("Dual Axis", "Line chart with dual y-axes.", MultiLineChartActivity.class);
        addExample("Inverted Axis", "Inverted y-axis.", InvertedLineChartActivity.class);
        addExample("Cubic", "Line chart with a cubic line shape.", CubicLineChartActivity.class);
        addExample("Colorful", "Colorful line chart.", LineChartActivityColored.class);
        addExample("Performance", "Render 30.000 data points smoothly.", PerformanceLineChart.class);
        addExample("Filled", "Colored area between two lines.", FilledLineActivity.class);

        ////
        addHeader("Bar Charts");

        addExample("Basic", "Simple bar chart.", BarChartActivity.class);
        addExample("Basic 2", "Variation of the simple bar chart.", AnotherBarActivity.class);
        addExample("Multiple", "Show multiple data sets.", BarChartActivityMultiDataset.class);
        addExample("Horizontal", "Render bar chart horizontally.", HorizontalBarChartActivity.class);
        addExample("Stacked", "Stacked bar chart.", StackedBarActivity.class);
        addExample("Negative", "Positive and negative values with unique colors.", BarChartPositiveNegative.class);
        addExample("Stacked 2", "Stacked bar chart with negative values.", StackedBarActivityNegative.class);
        addExample("Sine", "Sine function in bar chart format.", BarChartActivitySinus.class);

        ////
        addHeader("Pie Charts");

        addExample("Basic", "Simple pie chart.", PieChartActivity.class);
        addExample("Value Lines", "Stylish lines drawn outward from slices.", PiePolylineChartActivity.class);
        addExample("Half Pie", "180° (half) pie chart.", HalfPieChartActivity.class);

        ////
        addHeader("Other Charts");

        addExample("Combined Chart", "Bar and line chart together.", CombinedChartActivity.class);
        addExample("Scatter Plot", "Simple scatter plot.", ScatterChartActivity.class);
        addExample("Bubble Chart", "Simple bubble chart.", BubbleChartActivity.class);
        addExample("Candlestick", "Simple financial chart.", CandleStickChartActivity.class);
        addExample("Radar Chart", "Simple web chart.", RadarChartActivity.class);

        ////
        addHeader("Scrolling Charts");

        addExample("Multiple", "Various types of charts as fragments.", ListViewMultiChartActivity.class);
        addExample("View Pager", "Swipe through different charts.", SimpleChartDemo.class);
        addExample("Tall Bar Chart", "Bars bigger than your screen!", ScrollViewActivity.class);
        addExample("Many Bar Charts", "More bars than your screen can handle!", ListViewBarChartActivity.class);

        ////
        addHeader("Even More Line Charts");

        addExample("Dynamic", "Build a line chart by adding points and sets.", DynamicalAddingActivity.class);
        addExample("Realtime", "Add data points in realtime.", RealtimeLineChartActivity.class);
        addExample("Hourly", "Uses the current time to add a data point for each hour.", LineChartTime.class);
        //addItem("Realm.io Examples", "See more examples that use Realm.io mobile database.", RealmMainActivity.class, position++));


        MyAdapter adapter = new MyAdapter(this, mActivities);


        ListView lv = findViewById(R.id.listView1);
        lv.setAdapter(adapter);

        lv.setOnItemClickListener(this);
    }

    @Override
    public void onItemClick(AdapterView<?> av, View v, int pos, long arg3) {
        ContentItem example = mActivities.get(pos);
        if (example.klass == null)
            return;
        Intent intent = new Intent(this, mActivities.get(pos).klass);
        startActivity(intent);

        overridePendingTransition(R.anim.move_right_in_activity, R.anim.move_left_out_activity);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {

        Intent i;

        switch (item.getItemId()) {
            case R.id.viewGithub:
                i = new Intent(Intent.ACTION_VIEW);
                i.setData(Uri.parse("https://github.com/PhilJay/MPAndroidChart"));
                startActivity(i);
                break;
            case R.id.report:
                i = new Intent(Intent.ACTION_SENDTO, Uri.fromParts(
                        "mailto", "philjay.librarysup@gmail.com", null));
                i.putExtra(Intent.EXTRA_SUBJECT, "MPAndroidChart Issue");
                i.putExtra(Intent.EXTRA_TEXT, "Your error report here...");
                startActivity(Intent.createChooser(i, "Report Problem"));
                break;
            case R.id.website:
                i = new Intent(Intent.ACTION_VIEW);
                i.setData(Uri.parse("http://at.linkedin.com/in/philippjahoda"));
                startActivity(i);
                break;
        }

        return true;
    }

    private void addHeader(String name) {
        ContentItem item = new ContentItem(name, mPosition);
        mActivities.add(mPosition++, item);
    }

    private void addExample(String name, String description, Class klazz) {
        ContentItem item = new ContentItem(name, description, mPosition, klazz);
        mActivities.add(mPosition++, item);
    }
}

File: MPChartExample/src/main/java/com/xxmassdeveloper/mpchartexample/notimportant/MainActivity.kt
package com.xxmassdeveloper.mpchartexample.notimportant

import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.view.View
import android.view.WindowManager
import android.widget.AdapterView
import android.widget.AdapterView.OnItemClickListener
import android.widget.ListView
import androidx.appcompat.app.AppCompatActivity
import com.github.mikephil.charting.utils.Utils
import com.xxmassdeveloper.mpchartexample.*
import com.xxmassdeveloper.mpchartexample.fragments.ViewPagerSimpleChartDemo

class MainActivity : AppCompatActivity(), OnItemClickListener {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        window.setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN, WindowManager.LayoutParams.FLAG_FULLSCREEN)
        setContentView(R.layout.activity_main)

        // initialize the utilities
        Utils.init(this)
        val adapter = MyAdapter(this, menuItems)
        val lv = findViewById<ListView>(R.id.listView1)
        lv.adapter = adapter
        lv.onItemClickListener = this
    }

    override fun onItemClick(av: AdapterView<*>?, v: View, pos: Int, arg3: Long) {
        val intent = Intent(this, menuItems[pos].clazz)
        startActivity(intent)
        overridePendingTransition(R.anim.move_right_in_activity, R.anim.move_left_out_activity)
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        val i: Intent
        when (item.itemId) {
            R.id.viewGithub -> {
                i = Intent(Intent.ACTION_VIEW)
                i.data = Uri.parse("https://github.com/AppDevNext/AndroidChart")
                startActivity(i)
            }
            R.id.report -> {
                i = Intent(Intent.ACTION_SENDTO, Uri.fromParts(
                        "mailto", "philjay.librarysup@gmail.com", null))
                i.putExtra(Intent.EXTRA_SUBJECT, "MPAndroidChart Issue")
                i.putExtra(Intent.EXTRA_TEXT, "Your error report here...")
                startActivity(Intent.createChooser(i, "Report Problem"))
            }
            R.id.website -> {
                i = Intent(Intent.ACTION_VIEW)
                i.data = Uri.parse("http://at.linkedin.com/in/philippjahoda")
                startActivity(i)
            }
        }
        return true
    }

    companion object {
        val menuItems = ArrayList<ContentItem<out DemoBase>>().apply {
            add(0, ContentItem("Line Charts"))
            add(0, ContentItem("Line Charts"))
            add(1, ContentItem("Basic", "Simple line chart.", LineChartActivity1::class.java))
            add(2, ContentItem("Multiple", "Show multiple data sets.", MultiLineChartActivity::class.java))
            add(3, ContentItem("Dual Axis", "Line chart with dual y-axes.", LineChartDualAxisActivity::class.java))
            add(4, ContentItem("Inverted Axis", "Inverted y-axis.", InvertedLineChartActivity::class.java))
            add(5, ContentItem("Cubic", "Line chart with a cubic line shape.", CubicLineChartActivity::class.java))
            add(6, ContentItem("Colorful", "Colorful line chart.", LineChartActivityColored::class.java))
            add(7, ContentItem("Performance", "Render 30.000 data points smoothly.", PerformanceLineChart::class.java))
            add(8, ContentItem("Filled", "Colored area between two lines.", FilledLineActivity::class.java))

            add(9, ContentItem("Bar Charts"))
            add(10, ContentItem("Basic", "Simple bar chart.", BarChartActivity::class.java))
            add(11, ContentItem("Basic 2", "Variation of the simple bar chart.", AnotherBarActivity::class.java))
            add(12, ContentItem("Multiple", "Show multiple data sets.", BarChartActivityMultiDataset::class.java))
            add(13, ContentItem("Horizontal", "Render bar chart horizontally.", HorizontalBarChartActivity::class.java))
            add(14, ContentItem("Stacked", "Stacked bar chart.", StackedBarActivity::class.java))
            add(15, ContentItem("Negative", "Positive and negative values with unique colors.", BarChartPositiveNegative::class.java))
            //objects.add(16, ContentItem("Negative Horizontal", "demonstrates how to create a HorizontalBarChart with positive and negative values."))
            add(17, ContentItem("Stacked 2", "Stacked bar chart with negative values.", StackedBarActivityNegative::class.java))
            add(18, ContentItem("Sine", "Sine function in bar chart format.", BarChartActivitySinus::class.java))

            add(19, ContentItem("Pie Charts"))
            add(20, ContentItem("Basic", "Simple pie chart.", PieChartActivity::class.java))
            add(21, ContentItem("Value Lines", "Stylish lines drawn outward from slices.", PiePolylineChartActivity::class.java))
            add(22, ContentItem("Half Pie", "180° (half) pie chart.", HalfPieChartActivity::class.java))
            add(23, ContentItem("Specific positions", "This demonstrates how to pass a list of specific positions for lines and labels on x and y axis", SpecificPositionsLineChartActivity::class.java))

            add(24, ContentItem("Other Charts"))
            add(25, ContentItem("Combined Chart", "Bar and line chart together.", CombinedChartActivity::class.java))
            add(26, ContentItem("Scatter Plot", "Simple scatter plot.", ScatterChartActivity::class.java))
            add(27, ContentItem("Bubble Chart", "Simple bubble chart.", BubbleChartActivity::class.java))
            add(28, ContentItem("Candlestick", "Simple financial chart.", CandleStickChartActivity::class.java))
            add(29, ContentItem("Radar Chart", "Simple web chart.", RadarChartActivity::class.java))

            add(30, ContentItem("Scrolling Charts"))
            add(31, ContentItem("Multiple", "Various types of charts as fragments.", ListViewMultiChartActivity::class.java))
            add(32, ContentItem("View Pager", "Swipe through different charts.", ViewPagerSimpleChartDemo::class.java))
            add(33, ContentItem("Tall Bar Chart", "Bars bigger than your screen!", ScrollViewActivity::class.java))
            add(34, ContentItem("Many Bar Charts", "More bars than your screen can handle!", ListViewBarChartActivity::class.java))

            add(35, ContentItem("Even More Line Charts"))
            add(36, ContentItem("Dynamic", "Build a line chart by adding points and sets.", DynamicalAddingActivity::class.java))
            add(37, ContentItem("Realtime", "Add data points in realtime.", RealtimeLineChartActivity::class.java))
            add(38, ContentItem("Hourly", "Uses the current time to add a data point for each hour.", LineChartTime::class.java))
            //add(39, new ContentItem("Realm.io Examples", "See more examples that use Realm.io mobile database."));
        }
    }
}